### ISSUE TITLE
Design of UI and CardView

### ISSUE THREAD

**Issue by LeonYuAng3NT** [SmileWuji/tea-party-deception#4]
"In an abstract sense, the Tea Party Deception is more like a web application rather than a traditional PC software."

- I'm still not sure what exact application are we building here, but I assume it's a web application game.
So how do we config the front-end and back-end with the current design? This actually leads to my previous question about the server and it's better to have an API between UI and Server. We can decide this soon since I think there is no callback to the UI, but an IO between the host and Server.

- I think there is still a lot more to add for API if this is a web application game according to this:
https://developer.mozilla.org/en-US/docs/Games/Introduction. Another question would be, why do we use J2SE package based hierarchy? I don't see how this hierarchy applies to game development, but definitely a great pattern we should follow if we try to implement a software.

**Comment by SmileWuji** [comment]
You are right, there would be IO between the host and Ui... in fact, the IO is between server and Ui, as the host is entirely encapsulated by the server. #1 will try to eliminate the CardView since it is callback based and is confusing.

To split the responsibility, the host is the owner of a running instance of the game (i.e. the logic level), so it should only deal with the input and output (i.e. the datapath) related to the game. The server, on the other hand, is just an enhancement of the host. It provides connection to the Ui for sure. Or should I say if we split the data transmission modules out of the server then we don't need a server to play the game since it is logically not a dependency of the game itself?

The dual of "J2SE package based hierarchy" is the so-called "JigSaw modules" introduced (and is the standard) in Java 9. Mentioned in the documentation since the game only runs on Java 9 (it uses new APIs). At a later time, it is possible to explicitly make module files for the project.

Documentation will be updated based on the remarks in this issue.
